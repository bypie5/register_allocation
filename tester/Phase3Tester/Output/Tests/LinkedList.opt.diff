--- SelfTestCases/LinkedList.opt.out	2020-03-05 22:31:29.000000000 -0800
+++ ./Output/Tests/LinkedList.opt.out	2020-03-08 22:08:59.000000000 -0700
@@ -1,26 +1,3 @@
-25
-10000000
-39
-25
-10000000
-22
-39
-25
-1
-0
-10000000
-28
-22
-39
-25
-2220000
--555
--555
-28
-22
-25
-33300000
-22
-25
-44440000
-0
+--- Error (ProgramFault) ---
+require function address, found undefined value
+  Main() line 11
