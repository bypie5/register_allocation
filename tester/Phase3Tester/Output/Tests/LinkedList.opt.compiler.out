const empty_Element

const empty_List

const empty_LL

func Main [in 0, out 0, local 2]
local[0] = $s0
local[1] = $s1
$a0 = :empty_LL
call $s0
$s1 = $v0
PrintIntS($s1)
$s0 = local[0]
$s1 = local[1]
ret

func Element.Init [in 0, out 0, local 4]
local[0] = $s0
local[1] = $s1
local[2] = $s2
local[3] = $s3
$s2 = $a0
$s0 = $a1
$s1 = $a2
$s3 = $a3
[$s2+0] = $s0
[$s2+4] = $s1
[$s2+8] = $s3
$v0 = 1
$s0 = local[0]
$s1 = local[1]
$s2 = local[2]
$s3 = local[3]
ret

func Element.GetAge [in 0, out 0, local 2]
local[0] = $s0
local[1] = $s1
$s0 = $a0
$s1 = [$s0+0]
$v0 = $s1
$s0 = local[0]
$s1 = local[1]
ret

func Element.GetSalary [in 0, out 0, local 2]
local[0] = $s0
local[1] = $s1
$s0 = $a0
$s1 = [$s0+4]
$v0 = $s1
$s0 = local[0]
$s1 = local[1]
ret

func Element.GetMarried [in 0, out 0, local 2]
local[0] = $s0
local[1] = $s1
$s0 = $a0
$s1 = [$s0+8]
$v0 = $s1
$s0 = local[0]
$s1 = local[1]
ret

func Element.Equal [in 0, out 0, local 8]
local[0] = $s0
local[1] = $s1
local[2] = $s2
local[3] = $s3
local[4] = $s4
local[5] = $s5
local[6] = $s6
local[7] = $s7
$s3 = $a0
$s5 = $a1
$s6 = 1
if $s5 goto :null1
Error("null pointer")
null1:
$a0 = $s5
call $s0
$s7 = $v0
$s0 = [$s3+0]
$a0 = $s3
$a1 = $s7
$a2 = $s0
call $s2
$t0 = $v0
if $t0 goto :if1_else
$s6 = 0
goto :if1_end
if1_else:
if $s5 goto :null2
Error("null pointer")
null2:
$a0 = $s5
call $s1
$s0 = $v0
$s1 = [$s3+4]
$a0 = $s3
$a1 = $s0
$a2 = $s1
call $s2
$s7 = $v0
if $s7 goto :if2_else
$s6 = 0
goto :if2_end
if2_else:
$s0 = [$s3+8]
if0 $s0 goto :if3_else
if $s5 goto :null3
Error("null pointer")
null3:
$a0 = $s5
call $s4
$s0 = $v0
if $s0 goto :if4_else
$s6 = 0
goto :if4_end
if4_else:
$s0 = 0
if4_end:
goto :if3_end
if3_else:
if $s5 goto :null4
Error("null pointer")
null4:
$a0 = $s5
call $s4
$s1 = $v0
if0 $s1 goto :if5_else
$s6 = 0
goto :if5_end
if5_else:
$s0 = 0
if5_end:
if3_end:
if2_end:
if1_end:
$v0 = $s6
$s0 = local[0]
$s1 = local[1]
$s2 = local[2]
$s3 = local[3]
$s4 = local[4]
$s5 = local[5]
$s6 = local[6]
$s7 = local[7]
ret

func Element.Compare [in 0, out 0, local 5]
local[0] = $s0
local[1] = $s1
local[2] = $s2
local[3] = $s3
local[4] = $s4
$s0 = $a0
$s2 = $a1
$s1 = $a2
$s0 = 0
$s3 = Add($s1 1)
$s4 = LtS($s2 $s1)
if0 $s4 goto :if6_else
$s0 = 0
goto :if6_end
if6_else:
$s1 = LtS($s2 $s3)
if $s1 goto :if7_else
$s0 = 0
goto :if7_end
if7_else:
$s0 = 1
if7_end:
if6_end:
$v0 = $s0
$s0 = local[0]
$s1 = local[1]
$s2 = local[2]
$s3 = local[3]
$s4 = local[4]
ret

func List.Init [in 0, out 0, local 1]
local[0] = $s0
$s0 = $a0
[$s0+8] = 1
$v0 = 1
$s0 = local[0]
ret

func List.InitNew [in 0, out 0, local 4]
local[0] = $s0
local[1] = $s1
local[2] = $s2
local[3] = $s3
$s2 = $a0
$s1 = $a1
$s3 = $a2
$s0 = $a3
[$s2+8] = $s0
[$s2+0] = $s1
[$s2+4] = $s3
$v0 = 1
$s0 = local[0]
$s1 = local[1]
$s2 = local[2]
$s3 = local[3]
ret

func List.Insert [in 0, out 0, local 5]
local[0] = $s0
local[1] = $s1
local[2] = $s2
local[3] = $s3
local[4] = $s4
$s0 = $a0
$s2 = $a1
$s3 = $s0
$s0 = HeapAllocZ(12)
if $s0 goto :null5
Error("null pointer")
null5:
$a0 = $s0
$a1 = $s2
$a2 = $s3
$a3 = 0
call $s1
$s4 = $v0
$v0 = $s0
$s0 = local[0]
$s1 = local[1]
$s2 = local[2]
$s3 = local[3]
$s4 = local[4]
ret

func List.SetNext [in 0, out 0, local 2]
local[0] = $s0
local[1] = $s1
$s0 = $a0
$s1 = $a1
[$s0+4] = $s1
$v0 = 1
$s0 = local[0]
$s1 = local[1]
ret

func List.Delete [in 0, out 0, local 8]
local[0] = $s0
local[1] = $s1
local[2] = $s2
local[3] = $s3
local[4] = $s4
local[5] = $s5
local[6] = $s6
local[7] = $s7
$s0 = $a0
$s5 = $a1
$s7 = $s0
$t0 = 0
$t1 = Sub(0 1)
$t2 = $s0
$t3 = $s0
$t4 = [$s0+8]
$t5 = [$s0+0]
while1_top:
if $t4 goto :ss1_else
$s0 = Sub(1 $t0)
goto :ss1_end
ss1_else:
$s0 = 0
ss1_end:
if0 $s0 goto :while1_end
if $s5 goto :null6
Error("null pointer")
null6:
$a0 = $s5
$a1 = $t5
call $s1
$s0 = $v0
if0 $s0 goto :if8_else
$t0 = 1
$s0 = LtS($t1 0)
if0 $s0 goto :if9_else
if $t2 goto :null7
Error("null pointer")
null7:
$a0 = $t2
call $s6
$s7 = $v0
goto :if9_end
if9_else:
$s0 = Sub(0 555)
PrintIntS($s0)
if $t3 goto :null8
Error("null pointer")
null8:
if $t2 goto :null9
Error("null pointer")
null9:
$a0 = $t2
call $s6
$s0 = $v0
$a0 = $t3
$a1 = $s0
call $s3
$t6 = $v0
$s0 = Sub(0 555)
PrintIntS($s0)
if9_end:
goto :if8_end
if8_else:
$s0 = 0
if8_end:
if $t0 goto :if10_else
$t3 = $t2
if $t2 goto :null10
Error("null pointer")
null10:
$a0 = $t2
call $s6
$t2 = $v0
if $t2 goto :null11
Error("null pointer")
null11:
$a0 = $t2
call $s2
$t4 = $v0
if $t2 goto :null12
Error("null pointer")
null12:
$a0 = $t2
call $s4
$t5 = $v0
$t1 = 1
goto :if10_end
if10_else:
$s0 = 0
if10_end:
goto :while1_top
while1_end:
$v0 = $s7
$s0 = local[0]
$s1 = local[1]
$s2 = local[2]
$s3 = local[3]
$s4 = local[4]
$s5 = local[5]
$s6 = local[6]
$s7 = local[7]
ret

func List.Search [in 0, out 0, local 8]
local[0] = $s0
local[1] = $s1
local[2] = $s2
local[3] = $s3
local[4] = $s4
local[5] = $s5
local[6] = $s6
local[7] = $s7
$s0 = $a0
$s3 = $a1
$s6 = 0
$s7 = $s0
$t0 = [$s0+8]
$t1 = [$s0+0]
while2_top:
$s0 = Sub(1 $t0)
if0 $s0 goto :while2_end
if $s3 goto :null13
Error("null pointer")
null13:
$a0 = $s3
$a1 = $t1
call $s1
$s0 = $v0
if0 $s0 goto :if11_else
$s6 = 1
goto :if11_end
if11_else:
$s0 = 0
if11_end:
if $s7 goto :null14
Error("null pointer")
null14:
$a0 = $s7
call $s5
$s7 = $v0
if $s7 goto :null15
Error("null pointer")
null15:
$a0 = $s7
call $s2
$t0 = $v0
if $s7 goto :null16
Error("null pointer")
null16:
$a0 = $s7
call $s4
$t1 = $v0
goto :while2_top
while2_end:
$v0 = $s6
$s0 = local[0]
$s1 = local[1]
$s2 = local[2]
$s3 = local[3]
$s4 = local[4]
$s5 = local[5]
$s6 = local[6]
$s7 = local[7]
ret

func List.GetEnd [in 0, out 0, local 2]
local[0] = $s0
local[1] = $s1
$s0 = $a0
$s1 = [$s0+8]
$v0 = $s1
$s0 = local[0]
$s1 = local[1]
ret

func List.GetElem [in 0, out 0, local 2]
local[0] = $s0
local[1] = $s1
$s0 = $a0
$s1 = [$s0+0]
$v0 = $s1
$s0 = local[0]
$s1 = local[1]
ret

func List.GetNext [in 0, out 0, local 2]
local[0] = $s0
local[1] = $s1
$s0 = $a0
$s1 = [$s0+4]
$v0 = $s1
$s0 = local[0]
$s1 = local[1]
ret

func List.Print [in 0, out 0, local 8]
local[0] = $s0
local[1] = $s1
local[2] = $s2
local[3] = $s3
local[4] = $s4
local[5] = $s5
local[6] = $s6
local[7] = $s7
$s0 = $a0
$s5 = $s0
$s6 = [$s0+8]
$s7 = [$s0+0]
while3_top:
$s0 = Sub(1 $s6)
if0 $s0 goto :while3_end
if $s7 goto :null17
Error("null pointer")
null17:
$a0 = $s7
call $s1
$s0 = $v0
PrintIntS($s0)
if $s5 goto :null18
Error("null pointer")
null18:
$a0 = $s5
call $s4
$s5 = $v0
if $s5 goto :null19
Error("null pointer")
null19:
$a0 = $s5
call $s2
$s6 = $v0
if $s5 goto :null20
Error("null pointer")
null20:
$a0 = $s5
call $s3
$s7 = $v0
goto :while3_top
while3_end:
$v0 = 1
$s0 = local[0]
$s1 = local[1]
$s2 = local[2]
$s3 = local[3]
$s4 = local[4]
$s5 = local[5]
$s6 = local[6]
$s7 = local[7]
ret

func LL.Start [in 0, out 0, local 8]
local[0] = $s0
local[1] = $s1
local[2] = $s2
local[3] = $s3
local[4] = $s4
local[5] = $s5
local[6] = $s6
local[7] = $s7
$s0 = $a0
$s0 = HeapAllocZ(12)
if $s0 goto :null21
Error("null pointer")
null21:
$a0 = $s0
call $s1
$s7 = $v0
$t0 = $s0
if $t0 goto :null22
Error("null pointer")
null22:
$a0 = $t0
call $s1
$s7 = $v0
if $t0 goto :null23
Error("null pointer")
null23:
$a0 = $t0
call $s6
$s7 = $v0
$s0 = HeapAllocZ(12)
if $s0 goto :null24
Error("null pointer")
null24:
$a0 = $s0
$a1 = 25
$a2 = 37000
$a3 = 0
call $s3
$s7 = $v0
if $t0 goto :null25
Error("null pointer")
null25:
$a0 = $t0
$a1 = $s0
call $s4
$t0 = $v0
if $t0 goto :null26
Error("null pointer")
null26:
$a0 = $t0
call $s6
$s7 = $v0
PrintIntS(10000000)
$s0 = HeapAllocZ(12)
if $s0 goto :null27
Error("null pointer")
null27:
$a0 = $s0
$a1 = 39
$a2 = 42000
$a3 = 1
call $s3
$s7 = $v0
$s1 = $s0
if $t0 goto :null28
Error("null pointer")
null28:
$a0 = $t0
$a1 = $s0
call $s4
$t0 = $v0
if $t0 goto :null29
Error("null pointer")
null29:
$a0 = $t0
call $s6
$s7 = $v0
PrintIntS(10000000)
$s0 = HeapAllocZ(12)
if $s0 goto :null30
Error("null pointer")
null30:
$a0 = $s0
$a1 = 22
$a2 = 34000
$a3 = 0
call $s3
$s7 = $v0
if $t0 goto :null31
Error("null pointer")
null31:
$a0 = $t0
$a1 = $s0
call $s4
$t0 = $v0
if $t0 goto :null32
Error("null pointer")
null32:
$a0 = $t0
call $s6
$s7 = $v0
$t1 = HeapAllocZ(12)
if $t1 goto :null33
Error("null pointer")
null33:
$a0 = $t1
$a1 = 27
$a2 = 34000
$a3 = 0
call $s3
$s7 = $v0
if $t0 goto :null34
Error("null pointer")
null34:
$a0 = $t0
$a1 = $s1
call $s2
$t2 = $v0
PrintIntS($t2)
if $t0 goto :null35
Error("null pointer")
null35:
$a0 = $t0
$a1 = $t1
call $s2
$t2 = $v0
PrintIntS($t2)
PrintIntS(10000000)
$s0 = HeapAllocZ(12)
if $s0 goto :null36
Error("null pointer")
null36:
$a0 = $s0
$a1 = 28
$a2 = 35000
$a3 = 0
call $s3
$s7 = $v0
if $t0 goto :null37
Error("null pointer")
null37:
$a0 = $t0
$a1 = $s0
call $s4
$t0 = $v0
if $t0 goto :null38
Error("null pointer")
null38:
$a0 = $t0
call $s6
$s7 = $v0
PrintIntS(2220000)
if $t0 goto :null39
Error("null pointer")
null39:
$a0 = $t0
$a1 = $s1
call $s5
$t0 = $v0
if $t0 goto :null40
Error("null pointer")
null40:
$a0 = $t0
call $s6
$s7 = $v0
PrintIntS(33300000)
if $t0 goto :null41
Error("null pointer")
null41:
$a0 = $t0
$a1 = $s0
call $s5
$t0 = $v0
if $t0 goto :null42
Error("null pointer")
null42:
$a0 = $t0
call $s6
$s7 = $v0
PrintIntS(44440000)
$v0 = 0
$s0 = local[0]
$s1 = local[1]
$s2 = local[2]
$s3 = local[3]
$s4 = local[4]
$s5 = local[5]
$s6 = local[6]
$s7 = local[7]
ret

